tarantool/msgpack_ext/types/interval.py

```diff
@@ -0,0 +1,213 @@
+        # ---
+        # ...
+        #
+        # tarantool> dt2 = datetime.interval.new{year = 1, adjust='excess'}
+        # ---
+        # ...
+        #
+        # tarantool> (dt1 + dt2).adjust
+        # ---
+        # - 'cdata<enum 112>: 2'
+        # ...
+
+        return Interval(
+            year = self.year + other.year,
+            month = self.month + other.month,
+            day = self.day + other.day,
+            hour = self.hour + other.hour,
+            minute = self.minute + other.minute,
+            sec = self.sec + other.sec,
+            nsec = self.nsec + other.nsec,
+            adjust = self.adjust,
+        )
+
+    def __sub__(self, other):
+        if not isinstance(other, Interval):
+            raise TypeError(f"unsupported operand type(s) for -: '{type(self)}' and '{type(other)}'")
+
+        # Tarantool saves adjust of the first argument
+        #
+        # Tarantool 2.10.1-0-g482d91c66
+        #
+        # tarantool> dt1 = datetime.interval.new{year = 2, adjust='last'}
+        # ---
+        # ...
+        #
+        # tarantool> dt2 = datetime.interval.new{year = 1, adjust='excess'}
+        # ---
+        # ...
+        #
+        # tarantool> (dt1 - dt2).adjust
+        # ---
+        # - 'cdata<enum 112>: 2'
+        # ...
+
+        return Interval(
+            year = self.year - other.year,
+            month = self.month - other.month,
+            day = self.day - other.day,
+            hour = self.hour - other.hour,
+            minute = self.minute - other.minute,
+            sec = self.sec - other.sec,
+            nsec = self.nsec - other.nsec,
+            adjust = self.adjust,
+        )
+
+    def __eq__(self, other):
+        if not isinstance(other, Interval):
+            return False
+
+        # Tarantool interval compare is naive too
+        #
+        # Tarantool 2.10.1-0-g482d91c66
+        #
+        # tarantool> datetime.interval.new{hour=1} == datetime.interval.new{min=60}
+        # ---
+        # - false
+        # ...
+
+        for field_id in id_map.keys():
+            field_name = id_map[field_id]
+            if getattr(self, field_name) != getattr(other, field_name):
+                return False
+
+        return True
+
+    def __repr__(self):
+        return f'tarantool.Interval(year={self.year}, month={self.month}, day={self.day}, ' + \
+               f'hour={self.hour}, minute={self.minute}, sec={self.sec}, ' + \
+               f'nsec={self.nsec}, adjust={self.adjust})'
+
+    __str__ = __repr__
+
+    def msgpack_encode(self):
+        buf = bytes()
+
+        count = 0
+        for field_id in id_map.keys():
+            field_name = id_map[field_id]
+            value = getattr(self, field_name)
+
+            if field_name == 'adjust':
+                value = value.value
+
+            if value != 0:
+                buf = buf + msgpack.packb(field_id) + msgpack.packb(value)
+                count = count + 1
+
+        buf = msgpack.packb(count) + buf
```

Comment:
Up to you, but the class violates Single Responsibility Principle: it mixes encoding/decoding and the interval-logic code.

Reply:
I think we leave it as is for now since the design is the same as in `Datetime`